simulator-samples/sample-swagger/src/main/java/org/citrusframework/simulator/sample/SpecificPingResponseMessageBuilder.java

```diff
@@ -0,0 +1,140 @@
+     */
+    private static final TriFunction<OpenApiServerActionBuilder, OasOperation, HttpMessage, HttpServerResponseActionBuilder> NULL_RESPONSE = SpecificPingResponseMessageBuilder::createNull;
+
+    /**
+     * Map to store specific functions per ping id.
+     */
+    private static final Map<Integer, TriFunction<OpenApiServerActionBuilder, OasOperation, HttpMessage, HttpServerResponseActionBuilder>> SPECIFC_BUILDER_MAP = new HashMap<>();
+
+    // Specific responses for some ids, all others will be handled by returning null and letting the random generator do its work.
+    static {
+        SPECIFC_BUILDER_MAP.put(15000,
+            SpecificPingResponseMessageBuilder::createResponseWithDedicatedRequiredHeader);
+        SPECIFC_BUILDER_MAP.put(10000,
+            SpecificPingResponseMessageBuilder::createResponseWithMessageAndHeaders);
+        SPECIFC_BUILDER_MAP.put(5000, SpecificPingResponseMessageBuilder::createResponseWithSpecificBody);
+        SPECIFC_BUILDER_MAP.put(4000,
+            SpecificPingResponseMessageBuilder::createResponseWithRandomGenerationSuppressed);
+    }
+
+    @Override
+    public HttpServerResponseActionBuilder provideHttpServerResponseActionBuilder(
+        ScenarioRunner scenarioRunner, SimulatorScenario simulatorScenario,
+        HttpMessage receivedMessage) {
+
+        if (!(simulatorScenario instanceof HttpOperationScenario httpOperationScenario)) {
+            return null;
+        }
+
+        OpenApiServerActionBuilder openApiServerActionBuilder = new OpenApiActionBuilder(
+            httpOperationScenario.getOpenApiSpecification()).server(scenarioRunner.getScenarioEndpoint());
+
+        return SPECIFC_BUILDER_MAP.getOrDefault(getIdFromPingRequest(receivedMessage), NULL_RESPONSE).apply(openApiServerActionBuilder, httpOperationScenario.getOperation(), receivedMessage);
+    }
+
+    private static Integer getIdFromPingRequest(HttpMessage httpMessage) {
+        String uri = httpMessage.getUri();
+        Pattern pattern = Pattern.compile("/pingapi/v1/ping/(\\d*)");
+        Matcher matcher = pattern.matcher(uri);
+        if (matcher.matches()) {
+            return Integer.parseInt(matcher.group(1));
+        }
+        return MISSING_ID;
+    }
+
+    /**
+     * Sample to prove, that random data generation can be suppressed. Note that the generated
+     * response is thus invalid and will result in an error.
+     */
+    private static OpenApiServerResponseActionBuilder createResponseWithRandomGenerationSuppressed(
+        OpenApiServerActionBuilder openApiServerActionBuilder, OasOperation oasOperation,
+        HttpMessage receivedMessage) {
+        OpenApiServerResponseActionBuilder sendMessageBuilder = openApiServerActionBuilder.send(
+            oasOperation.operationId, "200").enableRandomGeneration(false);
+        sendMessageBuilder.message().body(format("{\"id\": %d, \"pingTime\": %d}",
+            getIdFromPingRequest(receivedMessage), System.currentTimeMillis()));
```

Comment:
```suggestion
            getIdFromPingRequest(receivedMessage), currentTimeMillis()));
```

please use static import for this.